From 91456b1bb8e93deef683a9c40b1fe7f1de3da99e Mon Sep 17 00:00:00 2001
From: Panu Matilainen <pmatilai@redhat.com>
Date: Tue, 25 Mar 2014 14:06:34 +0200
Subject: [PATCH] Ensure clean paths are used for matching in debugedit
 (RhBug:1077148)

- Things like extra slashes can cause sources not to be found at all
  due to path prefix matching, canonicalize the argument paths first.
---
 tools/debugedit.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tools/debugedit.c b/tools/debugedit.c
index df7becc..0f85885 100644
--- a/tools/debugedit.c
+++ b/tools/debugedit.c
@@ -1541,6 +1541,12 @@ main (int argc, char *argv[])
 	}
     }
 
+  /* Ensure clean paths, users can muck with these */
+  if (base_dir)
+    canonicalize_path(base_dir, base_dir);
+  if (dest_dir)
+    canonicalize_path(dest_dir, dest_dir);
+
   /* Make sure there are trailing slashes in dirs */
   if (base_dir != NULL && base_dir[strlen (base_dir)-1] != '/')
     {
-- 
1.9.0

